sdk/rust/src/api.rs

```diff
@@ -0,0 +1,503 @@
+use std::{collections::HashMap, env};
+
+use anyhow::{anyhow, Result};
+use log::{debug, info};
+use serde_json::Value;
+
+use crate::{
+    client::HttpClient,
+    models::{ExecutionProvider, FoundryListResponseModel, FoundryModelInfo},
+    service::{assert_foundry_installed, get_service_uri, start_service},
+};
+
+/// Manager for Foundry Local SDK operations.
+pub struct FoundryLocalManager {
+    service_uri: Option<String>,
+    client: Option<HttpClient>,
+    catalog_list: Option<Vec<FoundryModelInfo>>,
+    catalog_dict: Option<HashMap<String, FoundryModelInfo>>,
+    timeout: Option<u64>,
+}
+
+impl FoundryLocalManager {
+    /// Create a new FoundryLocalManager instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `alias_or_model_id` - Optional alias or model ID to download and load. Only used if bootstrap is true.
+    /// * `bootstrap` - Optional boolean to start the service if it is not running. Default is false.
+    /// * `timeout_secs` - Optional timeout for the HTTP client in seconds. Default is 10 seconds.
+    ///
+    /// # Returns
+    ///
+    /// A new FoundryLocalManager instance.
+    pub async fn new(
+        alias_or_model_id: Option<&str>,
+        bootstrap: Option<bool>,
```

Comment:
You don't seem to distinguish between `false` and `None` so perhaps `bootstrap: bool` will suffice and be simpler.